src/compiler/Stage/Optimize.elm

```diff
@@ -46,6 +47,26 @@ optimizePlus located =
             Nothing
 
 
+log : String -> a -> b -> b
+log label data returned =
+    Debug.log label data |> (\_ -> returned)
+
+
+optimizeCons : Typed.LocatedExpr -> Maybe Typed.LocatedExpr
+optimizeCons located =
+    case Typed.getExpr located of
+        Typed.Cons l r ->
+            case Typed.getExpr r of
+                Typed.List list ->
+                    Just (Typed.mapExpr (\_ -> Typed.List (l :: list)) r)
```

Comment:
I see you added `Typed.mapExpr`, and the optimization looks pretty nice with it. I know it's out of the scope of the PR, but can you please try doing the same for the `optimizePlus` function? It will be good to have them consistent.

Reply:
It's done! 👍